racket/collects/racket/port.rkt

```diff
@@ -164,6 +166,66 @@
 (define (call-with-input-bytes str proc)
   (with-input-from-x 'call-with-input-bytes 1 #t str proc))
 
+(define (combine-output-ports what . with)
+  (define port-list (cons what with))
+  (map (lambda (p)
+         (or (output-port? p)
+             (raise-argument-error 'combine-output-ports "output-port?" p))) port-list)
+  (define wrap-evt-with-evts
+    (lambda (evt . evts)
+      (if (equal? (length evts) 0)
+          evt
+          (wrap-evt evt (let ([wrapped-evts (apply wrap-evt-with-evts evts)])
+                          (lambda (v)
+                            (sync wrapped-evts)))))))
+  (make-output-port
+   'combined-output-port
+   ; evt
+   (apply wrap-evt-with-evts port-list)
+   ; write-out 
+   (lambda (s start end non-block? breakable?)
+     (let ([s (subbytes s start end)])
+       (if non-block?
+           (map (lambda (p)
```

Comment:
This indentation isn't what we'd normally use here -- if there are multiple lines, then the last argument should also go on its own line.